### PR TITLE
add a test to track inf due to fp16 overflow

### PR DESCRIPTION
**Pull request by liqiangxl**
When layer norm backward is segmented, the intermediate tensors between different fusion segments are saved as fp16. In a case with a hidden size of 27K, it leads to inf values. see https://github.com/NVIDIA/Fuser/issues/704
If the test case is using a dtype of fp16 and segmented, this PR scales down the inputs by a factor of 0.01.
It allows us to validate the fp16 segmented cases.

### CODE REVIEW

File: test/test_combined_inner_outer_reduction.cpp [hunk 1]
```diff
@@ -242,15 +253,38 @@ TEST_F(NVFuserTest, CombinedSchedulerLayerNormBackward_CUDA) {
...
   bool isBenchmark = false;
   bool onlyTestFirstCase = false;
   int verbose = 0;
   for (auto dtype : data_types) {
     for (auto batch_shape : batch_sizes) {
       for (auto norm_shape : hidden_sizes) {
-        std::tuple<float, float> avg_var =
-            runTest(batch_shape, norm_shape, dtype, isBenchmark, verbose);
+        bool scale_down_to_avoid_overflow = false;
+        if (dtype == DataType::Half && norm_shape[0] == 27648) {
```
Comment (naoyam) [inline test/test_combined_inner_outer_reduction.cpp:274]: Any problem if this is done always?

Comment (liqiangxl) [reply test/test_combined_inner_outer_reduction.cpp:274]: It still works. But sacling down inputs seems like sacling up tolerence. It may lose the capability to capture some bugs leading to only a small difference in the results.

Comment (naoyam) [reply test/test_combined_inner_outer_reduction.cpp:274]: How about scaling back the results?

File: test/test_combined_inner_outer_reduction.cpp [hunk 2]
```diff
@@ -165,12 +172,16 @@ TEST_F(NVFuserTest, CombinedSchedulerLayerNormBackward_CUDA) {
...
     }
-    TORCH_CHECK(
-        !fec.getMostRecentKernelRuntime()->isSegmented(),
-        "Fusion shouldn't be segmented! hidden size= ",
-        hidden_size,
-        ", dtype= ",
-        dtype == DataType::Float ? "Float." : "Half.");
+
+    if (dtype == DataType::Half) {
+      TORCH_CHECK(
```
Comment (naoyam) [inline test/test_combined_inner_outer_reduction.cpp:177]: nit: don't use TORCH_ in the tests anymore

File: test/test_combined_inner_outer_reduction.cpp [hunk 3]
```diff
@@ -165,12 +172,16 @@ TEST_F(NVFuserTest, CombinedSchedulerLayerNormBackward_CUDA) {
...
+          fec.getMostRecentKernelRuntime()->isSegmented() ==
```
Comment (naoyam) [inline test/test_combined_inner_outer_reduction.cpp:178]: Segmentation should be fine as long as the input size is not too large

File: test/test_combined_inner_outer_reduction.cpp (hunk 1 above)
Comment (liqiangxl) [reply test/test_combined_inner_outer_reduction.cpp]: Good idea! If scale back, the scale factor only has a small influence the max diff.
left: scale_factor = 1, right: scale_factor = 0.01
image

File: test/test_combined_inner_outer_reduction.cpp (hunk 2 above)
Comment (liqiangxl) [reply test/test_combined_inner_outer_reduction.cpp]: This check is removed.

File: test/test_combined_inner_outer_reduction.cpp (hunk 3 above)
Comment (liqiangxl) [reply test/test_combined_inner_outer_reduction.cpp]: This check is removed.

File: test/test_combined_inner_outer_reduction.cpp (hunk 1 above)
Comment (liqiangxl) [reply test/test_combined_inner_outer_reduction.cpp]: Great idea!  If scale back, the max_diff matains at same level whether with or without scale.
Without scale:
```
max_diff= 1.90735e-06
max_diff= 1.90735e-06
max_diff= 0.000244141
max_diff= 0.00012207
max_diff= 0.00012207
max_diff= 0.00195312
max_diff= 0.000244141
max_diff= 0.000488281
max_diff= 0.000976562
max_diff= 0.00195312
max_diff= inf
```
Scale by 0.01 then scale back by 1/0.01
```
max_diff= 0
max_diff= 3.8147e-06
max_diff= 0.00012207
max_diff= 0.000976562
max_diff= 0.00012207
max_diff= 0.000488281
max_diff= 0.000976562
max_diff= 0.000488281
max_diff= 0.00390625
max_diff= 0.00195312
max_diff= 0.0078125
```

**Review by naoyam — APPROVED** [review]
LGTM
